Added notebooks and action file for lakeFS hooks

vinodhini8694 opened the pull request:
Added notebooks, sample data file and action file for lakeFS hooks

kesarwam commented:
@vinodhini8694 I ran the Notebook and noticed few issues:

1. `client.branches.list_branches` is before `client.repositories.create_repository`, so list_branches command will always fail.
2. One of the cell refers to `temp.yaml` but this file doesn't exist in the repo/demo.
3. Quality check or hook is never invoked because `actions.yaml` file is not uploaded to repo.
4. Demo includes `netflix.csv` file but it is never used in the demo.
5. I will suggest to include few instructions also.
6. I will also suggest to clear outputs in the notebook.

vinodhini8694 commented:
Uploaded the notebook along with the README.

kesarwam commented:
Few issues I faced during testing this notebook:
1. AirPlay in MacOS Monterey uses port 5000 so I will suggest to use port 5001 for lakefs-hooks-server. I used port 5001 and it worked fine.
2. Version requirements for lakeFS-hooks should be resolved also: [](https://github.com/treeverse/lakeFS-hooks/pull/6)
3. Password for JupyterLabs UI is lakefs but it is not mentioned anywhere.
4. Everything Bagel Issue: Latest version of lakefs setup fails with following error: "Failed to setup DB: failed to open KV store: unknown driver". So, setup and example repo should be done manually otherwise this notebook will not run. So, I will suggest adding repo creation step in this notebook.

vinodhini8694 commented:
@kesarwam 

1. Modified port # to 5001 for hooks server.
2. Version requirements for lakeFS hooks requirements resolved: [PR](https://github.com/treeverse/lakeFS-hooks/pull/6)
3. Added jupyter notebook password in everything bagel docker-compose.yaml
4. This issue seems local to your setup. It was not reproducible in my local, the notebook runs fine and creates an example repo without any issue.

adipolak reviewed (approved):
approved merge.
